client/state/data-layer/wpcom-api-middleware.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import handlers from './wpcom';
+
+/**
+ * WPCOM Middleware API
+ *
+ * Intercepts actions requesting data provided by the
+ * WordPress.com API and passes them off to the
+ * appropriate handler.
+ */
+export const middleware = store => next => action =>
+	// we won't use has( handlers, action.type )
+	// here becuase of performance implications
```

Comment:
@dmsnell At this point, I'm poking at edge cases and micro-optimizations, so not terribly concerned, only intrigued 😄